Modules/Sources/WordPressUI/Views/Users/Views/UserDetailsView.swift

```diff
@@ -58,14 +66,16 @@ struct UserDetailsView: View {
                     Button(Strings.setNewPasswordActionTitle) {
                         presentPasswordAlert = true
                     }
-
-                    NavigationLink {
-                        // Pass this view's dismiss action, because if we delete a user, we want that screen *and* this one gone
-                        UserDeleteView(user: user, userProvider: userProvider, actionDispatcher: actionDispatcher, dismiss: dismissAction)
+                    Button(role: .destructive) {
+                        presentUserPicker = true
                     } label: {
-                        Text(Strings.deleteUserActionTitle)
-                            .foregroundStyle(Color.red)
+                        Text(
+                            deleteUserViewModel.isDeletingUser ?
```

Comment:
I'd suggest showing the "processing" state in the sheet and dismiss it only after the deletion is done.

Reply:
The sheet is dismiss-able by tapping the outside area. Also, the sheet is one of steps in the deleting process, for selecting a user to reassign content. After selecting a user, and the sheet is dismissed, there is a final confirmation alert. That's where we actually delete users.

<img width=400 src="https://github.com/user-attachments/assets/1bec58f1-d105-4658-916b-903a3b87a652" />